pkg/scalers/gcp_pub_sub_scaler.go

```diff
@@ -87,14 +108,23 @@ func parsePubSubMetadata(config *ScalerConfig) (*pubsubMetadata, error) {
 
 // IsActive checks if there are any messages in the subscription
 func (s *pubsubScaler) IsActive(ctx context.Context) (bool, error) {
-	size, err := s.GetSubscriptionSize(ctx)
-
-	if err != nil {
-		gcpPubSubLog.Error(err, "error getting Active Status")
-		return false, err
+	if s.metadata.mode == pubsubModeSubscriptionSize {
```

Comment:
Why not a switch here? It could ease the extendability of other metrics in case it's needed.

Reply:
good call, I'll fix it